### PR TITLE
[JIT] Remove BatchNorm and InstanceNorm special cases from AliasDB and replace with SchemaInfo is_mutable checks

### PR DESCRIPTION
**Pull request by ttanpcs**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82029
* #81917
* #81916
* #81837
* #81836
* #81786
* __->__ #81785

- Generalized AnalyzeImpl cases for batchNorm and InstanceNorm in alias_analysis.cpp using schema_info.
- Tested by ensuring all aliasDB special case checks for batchNorm and instanceNorm pass as expected.

### CODE REVIEW

File: torch/csrc/jit/ir/alias_analysis.cpp [hunk 1]
```diff
@@ -808,13 +804,26 @@ void AliasDb::analyzeImpl(Node* node) {
...
-
+  torch::utils::SchemaInfo schema_info(schema);
+  if (!isFrozen_ && node->hasNamedInput("training")) {
+    auto value = constant_as<bool>(node->namedInput("training"));
+    schema_info.addArgumentValue("training", !value.has_value() || *value);
+  } else if (!isFrozen_ && node->hasNamedInput("use_input_stats")) {
+    auto value = constant_as<bool>(node->namedInput("use_input_stats"));
+    schema_info.addArgumentValue(
+        "use_input_stats", !value.has_value() || *value);
+  }
```
Comment (davidberard98) [inline torch/csrc/jit/ir/alias_analysis.cpp:815]: actually do you think you could change the behavior of schema_info for this? I think this PR will technically work but on second thought I think schema_info isn't behaving correctly right now:

If we don't know the value of `training` (and we don't assign its value in schema_info) then I think we should conservatively assume that `batch_norm` is mutating inputs (in case training=True). But schemainfo right now will assume that training=False if we don't know its value, which means that if a user doesn't provide inputs then sometimes it will return incorrect mutation information.

And with this change I think this will need to be changed slightly to handle freezing differently

File: torch/csrc/jit/ir/alias_analysis.cpp [hunk 2]
```diff
@@ -805,14 +805,25 @@ void AliasDb::analyzeImpl(Node* node) {
...
-    schema_info.addArgumentValue(
-        "use_input_stats", !value.has_value() || *value);
+
+  // Add arguments for batch_norm and instance_norm special case.
+  if (node->hasNamedInput("training")) {
+    if (isFrozen_) {
+      schema_info.addArgumentValue("training", false);
+    } else {
+      auto value = constant_as<bool>(node->namedInput("training"));
+      schema_info.addArgumentValue("training", !value.has_value() || *value);
```
Comment (davidberard98) [inline torch/csrc/jit/ir/alias_analysis.cpp:815]: nit: in the case that we can't find the constant value (i.e. `!value.has_value()`) and `isFrozen_ == false` can we not assign a value to schema_info? Practically it's probably not an issue here, but hypothetically you might want to change the default behavior and in that case we don't want to add the argument value here.

**Review by davidberard98 — APPROVED** [review]
had a few nits, could you fix those?

Otherwise looks good, thanks for these changes!
